### ISSUE TITLE
Issue reproducing tokenized input for BERT

### ISSUE THREAD

**Issue by Muhtasham** [NVIDIA/TensorRT-LLM#1042]
### System Info

- GPU name: NVIDIA RTX A1000 
- Windows: 10
- 0.7.1 tensorrt-llm

### Who can help?

@byshiue @symphonylyh 

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

I'm utilizing BertTokenizer alongside optimized BERTModel. However, when decoding the output logits from the TensorRT-optimized BERT model, I encounter unexpected results, notably a significant presence of [unusedXXX] tokens in the decoded text, which suggests a potential mismatch between the tokenizer's vocabulary and the logits produced by the TensorRT engine.

Below is a simplified code snippet that illustrates the issue:

```python
from transformers import BertTokenizer

# Initialize the tokenizer
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')

# Example of encoding an input
input_text = "This is a test input."
encoded_input = tokenizer.encode_plus(input_text, return_tensors="pt")

# Pseudo-code: Assuming `logits` are obtained from the TensorRT-optimized model
logits = tensorrt_engine_infer(encoded_input['input_ids'])

# Decoding the output logits to text
decoded_output = tokenizer.decode(logits.argmax(-1), skip_special_tokens=True)
print("Decoded Output:", decoded_output)
```

Full code can be seen [here](https://github.com/Muhtasham/TensorRT-LLM/blob/fix/win-oom-a1000/examples/bert/run_encoded.py)

### Expected behavior

The expected behavior is that the decoded output closely resembles the input text or provides meaningful predictions for masked language modeling tasks, without an over-representation of [unusedXXX] tokens.

### actual behavior

After disabling Multi-Head Attention (MHA) kernels to sidestep potential accuracy issues with logits, I anticipated that the model's output would align closely with expected results. Instead, the decoded outputs primarily consist of [unusedXXX] tokens. This suggests a misalignment in how the tokenizer interprets the logits from the TensorRT-optimized BERT model, deviating from expected behavior where meaningful tokens are predicted.

### additional notes

- How can I refine the tokenization, encoding, and decoding workflow to ensure compatibility with a TensorRT-optimized BERT model, especially to mitigate the issue of [unusedXXX] tokens in the output?
- Are there best practices or adjustments recommended in the examples/bert to correctly encode and decode logits?
- Given the configurations in my config.json (provided below), have I overlooked any settings that could impact the tokenizer's performance with the TensorRT model?

My `config.json` below:

```json
{
  "builder_config": {
    "int8": false,
    "max_batch_size": 8,
    "max_input_len": 512,
    "name": "BertModel",
    "precision": "float16",
    "tensor_parallel": 1,
    "use_refit": false
  },
  "plugin_config": {
    "attention_qk_half_accumulation": false,
    "bert_attention_plugin": false,
    "context_fmha_type": 0,
    "gemm_plugin": false,
    "gpt_attention_plugin": false,
    "identity_plugin": false,
    "layernorm_plugin": false,
    "layernorm_quantization_plugin": false,
    "lookup_plugin": false,
    "lora_plugin": false,
    "multi_block_mode": false,
    "nccl_plugin": false,
    "paged_kv_cache": false,
    "quantize_per_token_plugin": false,
    "quantize_tensor_plugin": false,
    "remove_input_padding": false,
    "rmsnorm_plugin": false,
    "rmsnorm_quantization_plugin": false,
    "smooth_quant_gemm_plugin": false,
    "tokens_per_block": 0,
    "use_context_fmha_for_generation": false,
    "use_custom_all_reduce": false,
    "use_paged_context_fmha": false,
    "weight_only_groupwise_quant_matmul_plugin": false,
    "weight_only_quant_matmul_plugin": false
  }
}

```

I appreciate any advice or insights you can share to help resolve this issue. Thank you.

**Comment by juney-nvidia** [comment]
@Muhtasham  can you share the concrete command sequence to reproduce the issue? Including how you build the engine. 

Thanks
June

**Comment by Muhtasham** [comment]
> @Muhtasham  can you share the concrete command sequence to reproduce the issue? Including how you build the engine. 
> 
> 
> 
> Thanks
> 
> June

@juney-nvidia Hi there I just followed official Bert Example in TensorRT-LLM `rel` branch for [windows ](https://github.com/NVIDIA/TensorRT-LLM/tree/rel/examples/bert)

**Comment by Muhtasham** [comment]
@symphonylyh I tried to track down [here](https://github.com/NVIDIA/TensorRT-LLM/blob/2f169d17d5a2b7690f78deacedccb25831905e5c/tensorrt_llm/models/bert/model.py#L202), seems like initializing randomly?

**Comment by hello-11** [comment]
@Muhtasham Do you still have the problem? If not, we will close it soon.

**Comment by Muhtasham** [comment]
> @Muhtasham Do you still have the problem? If not, we will close it soon.

Yes, would be nice to have official documentation regarding this
